src/CommunityPatch/Patches/Perks/Intelligence/Engineering/ResolutePatch.cs

```diff
@@ -0,0 +1,142 @@
+      var perk = ActivePatch._perk;
+      
+      if (mobileParty.SiegeEvent == null) return;
+      var partyMemberBonus = GetPartyMemberBonus(mobileParty, perk);
+      if (partyMemberBonus < .01) return;
+
+      var partyMembersCount = (int) (partyMemberBonus / perk.PrimaryBonus);
+      var multiplicativeBonus = CalculateMultiplicativeBonus(perk.PrimaryBonus / 100, partyMembersCount);
+      var moraleLossReductionRate = multiplicativeBonus * -1;
+      
+      var totalMoraleLossReduction = CalculateTotalMoraleLossReduction(mobileParty, moraleLossReductionRate);
+      explanation?.AddLine(perk.Name.ToString(), totalMoraleLossReduction);
+      __result += totalMoraleLossReduction;
+    }
+    
+    private static float GetPartyMemberBonus(MobileParty mobileParty, PerkObject perk)
+    {
+      var perkPartyMemberValue = new ExplainedNumber(100);
+      PerkHelper.AddPerkBonusForParty(perk, mobileParty, ref perkPartyMemberValue);
+      var additiveMoralePenaltyReductionRate = perkPartyMemberValue.ResultNumber - perkPartyMemberValue.BaseNumber;
+      return additiveMoralePenaltyReductionRate;
+    }
+
+    private static float CalculateMultiplicativeBonus(float partyMemberBonus, int partyMemberCount) {
+      var finalRate = 0f;
+      var lastRate = 1f;
+      
+      for (var i = 0; i < partyMemberCount; i++) {
+        var nextRate = partyMemberBonus * lastRate;
+        finalRate += nextRate;
+        lastRate = nextRate;
+      }
+
+      return finalRate;
+    }
+
+    private static int CalculateTotalMoraleLossReduction(MobileParty mobileParty, float moraleLossReductionRate)
+    {
+      var recentEventsMoraleLossReduction = Math.Min(mobileParty.RecentEventsMorale, 0) * moraleLossReductionRate;
+      var starvationMoraleLossReduction = mobileParty.Party.IsStarving ? -30 * moraleLossReductionRate : 0;
+      var unpaidWageMoraleLossReduction = mobileParty.HasUnpaidWages > 0f ? -20 * moraleLossReductionRate : 0;
+      var partySizeMoraleLossReduction = CalculatePartySizeMoralePenalty(mobileParty) * moraleLossReductionRate;
+      var foodVarietyMoraleLossReduction = CalculateFoodVarietyPenalty(mobileParty) * moraleLossReductionRate;
+
+      var totalMoraleLossReduction = (int) (recentEventsMoraleLossReduction +
+        starvationMoraleLossReduction +
+        unpaidWageMoraleLossReduction +
+        partySizeMoraleLossReduction +
+        foodVarietyMoraleLossReduction);
+      
+      return totalMoraleLossReduction;
+    }
+    private static float CalculatePartySizeMoralePenalty(MobileParty party) {
+      var oversize = party.Party.NumberOfAllMembers - party.Party.PartySizeLimit;
+      if (oversize > 0) return -1f * (float) Math.Sqrt(oversize);
+      return 0f;
+    }
+
+    private static float CalculateFoodVarietyPenalty(MobileParty party) {
+      var variety = party.ItemRoster.FoodVariety;
+      if (variety <= 1) return -2;
+      if (variety <= 2) return -1;
+      return 0;
+    }
```

Comment:
Use the existing `CalculateFoodVarietyMoraleBonus` and `GetPartySizeMoraleEffect` functions instead.

Reply:
Just to clarify my understanding, is your suggestion for me to patch the following property/methods ? 
- MobileParty.RecentEventsMorale
- DefaultPartyMoraleModel.GetStarvationMoralePenalty()
- DefaultPartyMoraleModel.GetNoWageMoralePenalty()
- DefaultPartyMoraleModel.CalculateFoodVarietyMoraleBonus()
- DefaultPartyMoraleModel.GetPartySizeMoraleEffect()

What do you suggest me to do regarding the display of the Party Morale tooltip? I think it is good when we let the player know when an effect is active. Currently, I was displaying all the morale loss reduction sum up like this:

Starvation: -30
No wage: -20
Recent Events: -6
Resolute: +28.
----

By implementing the resolute perk on each property/method individually, I guess the final result look like this:
Starvation: -30
Resolute: +15
No Wage: -20
Resolute: +10
Recent events: -6
Resolute: +3

Is this acceptable or should I think hard about an alternative?

Reply:
StatExplainer handles calls of `AddLine` with the same `name` parameter. So if you call `explanation.AddLine("Resolute", 15f, StatExplainer.OperationType.Add)` and later `explanation.AddLine("Resolute", 3f, StatExplainer.OperationType.Add)`, when the text actually shows, it will be `Resolute: +18`.

Reply:
Awesome :)

Reply:
How would you access statExplainer from within MobileParty.RecentEventsMorale?

I guess patching DefaultPartyMoraleModel.GetEffectivePartyMorale is still necessary, right?

Reply:
hmm yeah, tricky. You could create a `_statExplainer` field like we do in other patches with `_perk`, but it's probably better to patch `GetEffectivePartyMorale`. Depends on how the code is looking, but either way is fine.

Reply:
I was working on refactoring the class but I noticed the methods DefaultPartyMoraleModel.GetStarvationMoralePenalty and DefaultPartyMoraleModel.GetNoWageMoralePenalty also don't receive the statExplainer. Do you still think it is worthy implementing a patch the other methods ?

Reply:
That was an option. Another one is patch only `GetEffectivePartyMorale`, but make sure to get the values of PartySizeMorale, FoodVariety, etc by calling the respective functions instead of rewriting them on the patch.